sdks/java/io/elasticsearch/contrib/ElasticsearchIOIT.java

```diff
@@ -0,0 +1,106 @@
+
+import static org.apache.beam.sdk.io.elasticsearch.ElasticsearchIOTest.ES_INDEX;
+import static org.apache.beam.sdk.io.elasticsearch.ElasticsearchIOTest.ES_TYPE;
+import static org.apache.beam.sdk.testing.SourceTestUtils.readFromSource;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Integration test for Elasticsearch IO.
+ */
+public class ElasticsearchIOIT {
+  private static final String ES_IP = "localhost";
+  private static final String ES_HTTP_PORT = "9200";
+
+  @Test
+  public void testSplitsVolume() throws Exception {
+    PipelineOptions options = PipelineOptionsFactory.create();
+    ElasticsearchIO.Read read =
+        ElasticsearchIO.read()
+            .withConnectionConfiguration(
+                ElasticsearchIO.ConnectionConfiguration
+                    .create("http://" + ES_IP + ":" + ES_HTTP_PORT, ES_INDEX, ES_TYPE));
+    ElasticsearchIO.BoundedElasticsearchSource initialSource =
+        new ElasticsearchIO.BoundedElasticsearchSource(read, null);
+    long desiredBundleSizeBytes = 0;
+    List<? extends BoundedSource<String>> splits = initialSource.splitIntoBundles(
+        desiredBundleSizeBytes, options);
+    SourceTestUtils.
+        assertSourcesEqualReferenceSource(initialSource, splits, options);
+    long expectedNbSplits = 5;
+    assertEquals(expectedNbSplits, splits.size());
+    int nonEmptySplits = 0;
+    for (BoundedSource<String> subSource : splits) {
+      if (readFromSource(subSource, options).size() > 0) {
+        nonEmptySplits += 1;
+      }
+    }
+    assertEquals(expectedNbSplits, nonEmptySplits);
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testReadVolume() throws Exception {
+    //use flink runner otherwise only one source is created
+    String[] args = new String[] { "--runner=FlinkRunner", "--project=test-project" };
+
+    TestPipeline pipeline =
+        TestPipeline.fromOptions(PipelineOptionsFactory.fromArgs(args).create());
+    PCollection<String> output = pipeline.apply(
+        ElasticsearchIO.read().withConnectionConfiguration(
+            ElasticsearchIO.ConnectionConfiguration
+                .create("http://" + ES_IP + ":" + ES_HTTP_PORT, ES_INDEX, ES_TYPE))
+            .withScrollKeepalive("5m"));
+    PAssert.thatSingleton(output.apply("Count", Count.<String>globally())).isEqualTo(96253L);
+    pipeline.run();
+  }
+
+  @Test
+  public void testEstimatedSizesVolume() throws IOException {
+    PipelineOptions options = PipelineOptionsFactory.create();
+    ElasticsearchIO.Read read =
+        ElasticsearchIO.read().withConnectionConfiguration(
+            ElasticsearchIO.ConnectionConfiguration
+              .create("http://" + ES_IP + ":" + ES_HTTP_PORT, ES_INDEX, ES_TYPE));
+    ElasticsearchIO.BoundedElasticsearchSource initialSource =
+        new ElasticsearchIO.BoundedElasticsearchSource(read, null);
+    assertTrue("Wrong estimated size", initialSource.getEstimatedSizeBytes
```

Comment:
assertThat(..., gt(1900000))

Also explain 1) why the size should be greater than that and 2) why that can be relied on so that the test is stable (does ElasticSearch guarantee that these statistics are always up-to-date?)